### PR TITLE
Fix bug in `validate_fft_implicit_solver_grid `

### LINKED ISSUE #2696: Bug in `validate_fft_implicit_solver_grid`?
I believe this 

https://github.com/CliMA/Oceananigans.jl/blob/67ba7f86099c9d206a20dcaa1e640bcf8da04ead/src/Models/HydrostaticFreeSurfaceModels/fft_based_implicit_free_surface_solver.jl#L24

should be

```Julia
validate_fft_implicit_solver_grid(ibg.underlying_grid) 
```

cc @glwagner, @simone-silvestri

### PR DESCRIPTION
**Pull request by navidcy**
Closes #2696

### CODE REVIEW

**Review by tomchor — APPROVED** [review]
Nice!

File: src/Models/HydrostaticFreeSurfaceModels/mg_implicit_free_surface_solver.jl [hunk 1]
```diff
@@ -55,7 +56,7 @@ for a fluid with variable depth `H`, horizontal areas `Az`, barotropic volume fl
 step `Δt`, gravitational acceleration `g`, and free surface at the `n`-th time-step `ηⁿ`.
 """
 function MGImplicitFreeSurfaceSolver(grid::AbstractGrid, 
-                                     settings, 
+                                     settings=nothing, 
                                      gravitational_acceleration = nothing,
```
Comment (navidcy) [inline src/Models/HydrostaticFreeSurfaceModels/mg_implicit_free_surface_solver.jl:60]:
```suggestion
                                     gravitational_acceleration = g_Earth, 
```
?

File: validation/implicit_free_surface/implicit_surface_solvers_benchmark.jl [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+const Lz = underlying_grid.Lz
+const width = 50kilometers
+bump(x, y) = - Lz * (1 - 0.2 * exp(-(x^2 + y^2) / 2width^2))
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(bump))
+
+# this is to fix a bug in validate_fft_implicit_solver_grid
+import Oceananigans.Models.HydrostaticFreeSurfaceModels.validate_fft_implicit_solver_grid
+validate_fft_implicit_solver_grid(ibg::ImmersedBoundaryGrid) = validate_fft_implicit_solver_grid(ibg.underlying_grid)
+
+# fft_preconditioner = FFTImplicitFreeSurfaceSolver(grid)
```
Comment (francispoulin) [inline validation/implicit_free_surface/implicit_surface_solvers_benchmark.jl:25]: Is the plan to keep these comments in the merge?

File: validation/implicit_free_surface/implicit_surface_solvers_benchmark.jl [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ϵb = 1e-2 * δb # noise amplitude
+
+bᵢ(x, y, z) = N² * z + δb * ramp(y, δy) + ϵb * randn()
+
+set!(model, b=bᵢ)
+
+Δt = 10minutes
+simulation = Simulation(model; Δt, stop_time=2days)
+
+# wizard = TimeStepWizard(cfl=0.2, max_change=1.1, max_Δt=simulation.Δt)
```
Comment (francispoulin) [inline validation/implicit_free_surface/implicit_surface_solvers_benchmark.jl:80]: Again, do you want to keep these comments?

**Review by francispoulin — APPROVED** [review]

File: validation/implicit_free_surface/implicit_surface_solvers_benchmark.jl (hunk 2 above)
Comment (navidcy) [reply validation/implicit_free_surface/implicit_surface_solvers_benchmark.jl]: I cleaned it up

**Pull request merged**